From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Mon, 26 Mar 2018 14:20:31 +0200
Subject: [PATCH] rbd: fix cache mode behavior

Either the cache mode asks for a cache or not. There's no
point in having a "temporary" cache mode. This option AFAIK
was introduced as a hack for ancient virtio drivers. If
anything, we should have a separate option for it. Better
yet, VMs affected by the related issue should simply
explicitly choose writethrough.

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 block/rbd.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/block/rbd.c b/block/rbd.c
index c4732a4b12..0374f3db27 100644
--- a/block/rbd.c
+++ b/block/rbd.c
@@ -643,9 +643,7 @@ static int qemu_rbd_open(BlockDriverState *bs, QDict *options, int flags,
         rados_conf_set(s->cluster, "rbd_cache", "true");
     }
 
-    if (flags & BDRV_O_NO_FLUSH) {
 	rados_conf_set(s->cluster, "rbd_cache_writethrough_until_flush", "false");
-    }
 
     r = rados_connect(s->cluster);
     if (r < 0) {
-- 
2.11.0

